lightning-graph-sync/src/processing.rs

```diff
@@ -0,0 +1,591 @@
+
+	// shadow variable for ownership release after length-restricted reads
+	let mut read_cursor = read_cursor;
+
+	let chain_hash: BlockHash = Readable::read(read_cursor)?;
+	let latest_seen_timestamp: u32 = Readable::read(read_cursor)?; //
+	let emulated_last_seen_timestamp = u32::max(latest_seen_timestamp, retrogressed_timestamp);
+
+	let mut node_ids: Vec<PublicKey> = vec![];
+	let node_id_count: u32 = Readable::read(read_cursor)?;
+	for _ in 0..node_id_count {
+		let current_node_id = Readable::read(read_cursor)?;
+		node_ids.push(current_node_id);
+	}
+
+	let mut previous_scid = None;
+	let announcement_count: u32 = Readable::read(read_cursor)?;
+	for _ in 0..announcement_count {
+		let features = Readable::read(read_cursor)?;
+
+		// handle SCID
+		let short_channel_id = if let Some(previous_scid) = previous_scid {
+			let scid_delta: BigSize = Readable::read(read_cursor)?;
+			previous_scid + scid_delta.0
+		} else {
+			Readable::read(read_cursor)?
+		};
+		previous_scid = Some(short_channel_id);
+
+		// let short_channel_id = Readable::read(read_cursor)?;
+		let node_id_1_index: BigSize = Readable::read(read_cursor)?;
+		let node_id_2_index: BigSize = Readable::read(read_cursor)?;
+		let node_id_1 = node_ids[node_id_1_index.0 as usize];
+		let node_id_2 = node_ids[node_id_2_index.0 as usize];
+
+		let announcement_result = network_graph.update_channel_from_partial_announcement(
+			short_channel_id,
+			emulated_last_seen_timestamp as u64,
+			features,
+			node_id_1,
+			node_id_2,
+		);
+		if let Err(lightning_error) = announcement_result {
+			if let ErrorAction::IgnoreDuplicateGossip = lightning_error.action {
+				// everything is fine, just a duplicate channel announcement
+			} else {
+				return Err(lightning_error.into());
+			}
+		}
+	}
+
+	previous_scid = None; // updates start at a new scid
+
+	let update_count: u32 = Readable::read(read_cursor)?;
+	if update_count < 1 {
+		return Ok(());
+	}
+	let non_incremental_update_presence_flag: u8 = Readable::read(read_cursor)?;
+	let has_non_incremental_updates = (non_incremental_update_presence_flag == 1);
+	// without non-incremental updates, the following fields cannot even be calculated by the server
+	let default_cltv_expiry_delta: u16 = if has_non_incremental_updates {
+		Readable::read(&mut read_cursor)?
+	} else {
+		0
+	};
+	let default_htlc_minimum_msat: u64 = if has_non_incremental_updates {
+		Readable::read(&mut read_cursor)?
+	} else {
+		0
+	};
+	let default_fee_base_msat: u32 = if has_non_incremental_updates {
+		Readable::read(&mut read_cursor)?
+	} else {
+		0
+	};
+	let default_fee_proportional_millionths: u32 = if has_non_incremental_updates {
+		Readable::read(&mut read_cursor)?
+	} else {
+		0
+	};
+	let tentative_default_htlc_maximum_msat: u64 = if has_non_incremental_updates {
+		Readable::read(&mut read_cursor)?
+	} else {
+		0
+	};
+	let default_htlc_maximum_msat = if tentative_default_htlc_maximum_msat == u64::MAX {
```

Comment:
Note that `u64::MAX` is only available as of 1.43 so you may need to use `u64::max_value()` here and elsewhere.

Reply:
this doesn't appear to be tripping up any CI tests, and max_value is deprecated. I'll look into why the CI tests aren't failing, because they should.

Reply:
Hmm, may need to update `.github/workflows/build.yml` to include the new crate.

Reply:
Spot on

Reply:
yup, looks like you need to use `u64::max_value()`